src/helpers/indexing-helper.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Yoast\WP\SEO\Helpers;
+
+/**
+ * A helper object for indexing.
+ */
+class Indexing_Helper {
+
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	protected $options_helper;
+
+	/**
+	 * Indexing_Helper constructor.
+	 *
+	 * @param Options_Helper $options_helper The options helper.
+	 */
+	public function __construct( Options_Helper $options_helper ) {
+		$this->options_helper = $options_helper;
+	}
+
+	/**
+	 * Sets the indexing reason.
+	 *
+	 * @param string $reason The indexing reason.
+	 *
+	 * @return void
+	 */
+	public function set_reason( $reason ) {
+		$this->options_helper->set( 'indexing_reason', $reason );
+	}
+
+	/**
+	 * Gets the indexing reason.
+	 *
+	 * @return mixed Returns the reason if found, and an empty string otherwise.
```

Comment:
I think this will always return a string. Either way, we want to be as specific as possible, so for example `string|null` would be better than `mixed` 🙂.
```suggestion
	 * @return string Returns the reason if found, and an empty string otherwise.
```

Reply:
Good point!